### DISCUSSION TITLE
Yocto hss generator yaml lib missing

### DISCUSSION THREAD

**Discussion by Micko8** [polarfire-soc/polarfire-soc-documentation#228] [Linux build systems for PolarFire SoC] [answered]
Hello,
I want to build mpfs-dev-cli for Aries Kit : [https://github.com/ARIES-Embedded/meta-polarfire-soc-yocto-bsp](url)
I'm facing this issue regarding yaml lib at 70% completion of "MACHINE=m100pfsevp bitbake mpfs-dev-cli"
Basiccaly ld command can't find -lyaml .

Please see the error message below.
[log.txt](https://github.com/polarfire-soc/polarfire-soc-documentation/files/9851915/log.txt)

I am pretty sure that it is related to my hss.bb recipe.
I tried to compare and use the one from polarfire soc yocto for icicle bsp from here [https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp/blob/master/recipes-bsp/hss-payload-generator-native/hss-payload-generator-native.bb](url)  but I don't know what to do except add  "inherit native" in hss.bb, which is not the solution apparently.
I don't understand because this yaml lib seems to be present in files.
I'm new in Yocto project and bitbake ... Can someone help me please ? 
Thank you

**Comment by hughbreslin** [comment, marked as answer]
Hey @Micko8 we can't give you a huge amount of support as this isn't part of our org... You'd be better suited to reach out to Aries for support.

Can I just check you saw the dependences at the end of the readme in the "Host PC setup for Yocto" and installed `libyaml-dev`?

**Reply by Micko8** [reply]
Thank you @hughbreslin 
Indeed I have doubled check the dependencies needed and I missed to add those ones. It's because when I saw "Other Dependencies" I did not want to install more than needed ... But obviously I was wrong.
